src/blob/blob_client.cpp

```diff
@@ -196,12 +196,13 @@ std::future<storage_outcome<container_property>> blob_client::get_container_prop
             container_property properties(true);
             properties.etag = http->get_response_header(constants::header_etag);
 
-            auto& headers = http->get_response_headers();
-            for (auto iter = headers.begin(); iter != headers.end(); ++iter)
+            const auto& headers = http->get_response_headers();
+            properties.metadata.reserve(headers.size());
+            for (const auto& header : headers)
             {
-                if (iter->first.find(constants::header_ms_meta_prefix) == 0)
+                if (header.first.find(constants::header_ms_meta_prefix) == 0)
                 {
-                    properties.metadata.push_back(std::make_pair(iter->first.substr(constants::header_ms_meta_prefix_size), iter->second));
+                    properties.metadata.emplace_back(header.first.substr(constants::header_ms_meta_prefix_size), header.second);
                 }
             }
```

Comment:
This code is duplicated with `get_blob_properties`, it would be better to abstract the logic and make it a function, for future use in file service.

Reply:
@katmsft I'm not sure I agree with you. Since this piece of code is processing properties and metadata. Properties for file/blob container/blob might be different. So I think it's also okay to leave it as it is.

Reply:
The best solution for this is to use a transform_if algorithm, but it is not in the STL.
I agree that we should eliminate duplication, but where do I put this function?

Reply:
@JinmingHu-MSFT You are absolutely right about the properties being different. What I was suggesting was the logic of extracting metadata from the headers. This is universal for blob/container/file/share/directory, potentially there will be 5 places that shares the same code logic.
@AlexandreBaiao I think it would be nice to have it in util.h so that if in the future file service is supported in CPPLite, it can be also called when extracting the metadata.